src/engine/engine.ts

```diff
@@ -278,3 +282,20 @@ async function getRemoteUrl(options) {
   const git = new Git(process.cwd(), options.git);
   return await git.getRemoteUrl(options.remote);
 }
+
+function tryParseGhPagesUrl(options: any): any {
```

Comment:
Options and return value shouldn't be typeof `any`

Reply:
Than you for the review.
[Type of `options`](https://github.com/angular-schule/angular-cli-ghpages/blob/master/src/engine/engine.ts#L43+L45
) is actually too complex to define in a function.

Instead of adding tryParseGhPagesUrl,  it is enough to parse url in `prepareOptions` function.
What do you think?

```ts
if (options.repo) {
    // Not assume custom domain page
    let trimEndDotGit = options.repo
      .replace(/\/\s*$/, '')
      .replace(/\.git\s*$/, '');

    const matchEndsWithRepoName = trimEndDotGit.match(/github.com(\/|:)(.*)\/(.*)$/);
    if (matchEndsWithRepoName) {
      options.ghPagesUrl = `https://${matchEndsWithRepoName[2]}.github.io/${matchEndsWithRepoName[3]}`;
    }
}

return options;
```